src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -157,7 +163,26 @@ protected IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCulture
             foreach (var name in resourceNames)
             {
                 var value = GetStringSafely(name, culture);
-                yield return new LocalizedString(name, value ?? name, resourceNotFound: value == null);
+                yield return new LocalizedString(name, value, resourceNotFound: value == null);
+            }
+        }
+
+        private string ApplyResourceLookupBehavior(string name, string value)
+        {
+            switch(_resourceLookupBehavior)
+            {
+                case ResourceLookupBehavior.ThrowIfNotFound:
+                    if(value == null)
+                    {
+                        throw new ArgumentOutOfRangeException(
+                            nameof(name),
+                            name,
+                            "The name of the string resource is missing from the resource source.");
```

Comment:
Doesn't seem like the right exception to throw. Probably should throw a new custom exception, e.g. `LocalizedStringNotFoundException`. The error message will need to be localized with traditional resources, e.g. https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Resources.resx